api/v1beta1/dataplane_types.go

```diff
@@ -32,6 +34,7 @@ func init() {
 // +kubebuilder:object:root=true
 // +kubebuilder:subresource:status
 // +kubebuilder:resource:shortName=kodp,categories=kong;all
+// +kubebuilder:validation:XValidation:message="Extension not allowed for DataPlane",rule="has(self.spec.extensions) ? self.spec.extensions.all(e, e.group == 'gateway-operator.konghq.com' && e.kind == 'DataPlaneKonnectExtension') : true"
```

Comment:
Should the message be `Extension other then Konnect not allowed for DataPlane` here?

Reply:
I'd keep this as we can add other extensions in the future and don't bother ourselves with changing this error message.